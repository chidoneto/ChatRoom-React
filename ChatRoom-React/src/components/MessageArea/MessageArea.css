.MessageArea {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    background-color: white;
    padding: 27px 0;
    position: sticky;
    -webkit-box-shadow: 0 -3px 4px 0 rgba(0,0,0,.2);
    box-shadow: 0 -3px 4px 0 rgba(0,0,0,.2);
}

.MessageArea input {
    margin: 2px 1% 0;
    margin-left: calc(3% + 2px);
    width: 78%;
    height: 20px;
    padding: 15px 3%;
    font-size: 100%;
    border: 1px solid #EBEDEF;
    border-radius: 5px;
    box-shadow: 0 0 1px 1px lightgrey;
    overflow: scroll;
}

.MessageArea input:focus, textarea:focus, select:focus{
    outline: none;
}

.MessageArea button {
    height: 52px;
    margin: 2px 3% 0 0px;
    padding: 15px 2%;
    font-size: 110%;
    font-weight: 700;
    border: 0;
    border-radius: 5px;
    color: #3B7ADB;
    background-color: white;
    transition: background-color 0.3s, border 0.3s, color 0.3s;
}

.MessageArea button:hover {
    border-radius: 5px;
    color: white;
    background-color: #3B7ADB;
    cursor: pointer;
}

.MessageArea button:disabled {
    text-decoration: line-through;
}

.MessageArea button:disabled:hover {
    background-color: lightgray;
}

@media (max-width: 480px) {
    .MessageArea {
        padding: 23px 0;
    }
    .MessageArea input {
        height: 18px;
        font-size: 95%;
    }
    .MessageArea button {
        height: 47px;
        margin: 3px 3% 0 0px;
        padding: 15px 2%;
        font-size: 100%;
    }
}