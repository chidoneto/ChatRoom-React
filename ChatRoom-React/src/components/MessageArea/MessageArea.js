import React, { useState, useRef, useEffect } from 'react';
import { PropTypes } from 'prop-types';

import './MessageArea.css';


const MessageArea = (props) => {
  const { className, placeHolder, actionText, onMessageTyped, disabled } = props;

  const classes = `MessageArea ${className}`;
  const [ messageText, setMessageText ] = useState('');
  const inputRef = useRef();

  const submitHandler = (event) => {
    event && event.preventDefault();
    onMessageTyped(messageText);
    setMessageText('');
  }

  useEffect(() => {
    inputRef.current.focus();
  }, [disabled]);

  return (
    <form className={classes} onSubmit={submitHandler} >
      <input ref={inputRef}
        placeholder={placeHolder || "Type a message..."}
        value={messageText} 
        onChange={event => setMessageText(event && event.target.value)}
        disabled={disabled} 
      >
      </input>
      <button disabled={disabled} >{actionText || "Send"}</button>
    </form>
  );
};

MessageArea.propTypes = {
  onMessageTyped: PropTypes.func.isRequired,
  disabled: PropTypes.bool
};

export default MessageArea;