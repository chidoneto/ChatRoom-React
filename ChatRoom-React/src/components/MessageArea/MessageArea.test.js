import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MessageArea from './MessageArea.js';
 
configure({adapter: new Adapter()});
 
describe('<MessageArea /> with no props', () => {
    let messageValue;
    const mockCallback = jest.fn((message) => { messageValue = message; });
    let wrapper;
    
    beforeEach(() => { wrapper = shallow(<MessageArea onMessageTyped={mockCallback}/>); });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have an input and a button', () => {
        expect(wrapper.find('.MessageArea')).toHaveLength(1);
        expect(wrapper.find('input')).toHaveLength(1);
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('should trigger callback on button click', () => {
        wrapper.find('input').simulate('change');
        expect(mockCallback).toHaveBeenCalledTimes(0);
    });

    it('should set a message, and receive such value back on submit', () => {
        const MESSAGE = 'This is just a sample message text. No more, no less...';

        wrapper.find('input').simulate('change', { target: { value:  MESSAGE} });
        expect(wrapper.find('input').prop('value')).toEqual(MESSAGE);
        wrapper.find('.MessageArea').simulate('submit');
        expect(messageValue).toEqual(MESSAGE);
    });

    it('input and submit button should be disabled when called that way', () => {
        wrapper.setProps({ disabled: true });
        expect(wrapper.find('input').prop('disabled')).toBeTruthy();
        expect(wrapper.find('button').prop('disabled')).toBeTruthy();
    });

    it('input and submit button should be enabled when called that way', () => {
        wrapper.setProps({ disabled: false });
        expect(wrapper.find('input').prop('disabled')).toBeFalsy();
        expect(wrapper.find('button').prop('disabled')).toBeFalsy();
    });
});
  
