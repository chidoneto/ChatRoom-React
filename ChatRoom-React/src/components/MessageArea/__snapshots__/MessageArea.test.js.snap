// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageArea /> with no props should match the snapshot 1`] = `ShallowWrapper {}`;
